src/interpreter.jl

```diff
@@ -35,6 +35,14 @@ function evaluate_with_input(tab::SymbolTable, expr::Any, input::Dict)
 end
 
 
+"""
+Executes a given expression on a set of inputs and returns the respective outputs.
+"""
+function execute_program_on_examples(tab::Symboltable, expr::Any, example_inputs::Vector{Any})::Vector{Any}
```

Comment:
Inputs can be `Vector{Dict{Symbol, Any}}` instead of `Vector{Any}`. Inputs are always defined as a dictionary with variable assignments.

Reply:
Then we should also change https://github.com/Herb-AI/Data.jl/blob/master/src/Data.jl#L22 ?

Reply:
Good catch, I didn't know about that. I think we should change it, so I made a PR:

- https://github.com/Herb-AI/Data.jl/pull/2